src/JuMP.jl

```diff
@@ -178,12 +179,16 @@ mutable struct Model <: AbstractModel
     end
 end
 
+# In Automatic and Manual mode, `m.moibackend` is a `LazyBridgeOptimizer` and
+# the `CachingOptimizer` is stored in the `model` field
+caching_optimizer(m::Model) = m.moibackend.model
```

Comment:
This method should assert Automatic or Manual mode. It's invalid and potentially dangerous in direct mode.